guides/source/testing.md

```diff
@@ -859,292 +846,440 @@ users(:david, :steve)
+the right response body has been generated.
 
-```ruby
-require "test_helper"
+The `get` method kicks off the web request and populates the results into the
+`@response`. It can accept up to 6 arguments:
 
-class ApplicationSystemTestCase < ActionDispatch::SystemTestCase
-  driven_by :selenium, using: :firefox
-end
-```
+* The URI of the controller action you are requesting. This can be in the form
+  of a string or a route helper (e.g. `articles_url`).
+* `params`: option with a hash of request parameters to pass into the action
+  (e.g. query string parameters or article variables).
+* `headers`: for setting the headers that will be passed with the request.
+* `env`: for customizing the request environment as needed.
+* `xhr`: whether the request is Ajax request or not. Can be set to true for
+  marking the request as Ajax.
+* `as`: for encoding the request with different content type.
 
-If you want to use a headless browser, you could use Headless Chrome or Headless Firefox by adding
-`headless_chrome` or `headless_firefox` in the `:using` argument.
+All of these keyword arguments are optional.
 
-```ruby
-require "test_helper"
+Example: Calling the `:show` action for the first `Article`, passing in an
+`HTTP_REFERER` header:
 
-class ApplicationSystemTestCase < ActionDispatch::SystemTestCase
-  driven_by :selenium, using: :headless_chrome
-end
+```ruby
+get article_url(Article.first), headers: { "HTTP_REFERER" => "http://example.com/home" }
 ```
 
-If you want to use a remote browser, e.g.
-[Headless Chrome in Docker](https://github.com/SeleniumHQ/docker-selenium),
-you have to add remote `url`  and set `browser` as remote through `options`.
+Another example: Calling the `:update` action for the last `Article`, passing in
+new text for the `title` in `params`, as an Ajax request:
 
 ```ruby
-require "test_helper"
-
-class ApplicationSystemTestCase < ActionDispatch::SystemTestCase
-  url = ENV.fetch("SELENIUM_REMOTE_URL", nil)
-  options = if url
-    { browser: :remote, url: url }
-  else
-    { browser: :chrome }
-  end
-  driven_by :selenium, using: :headless_chrome, options: options
-end
+patch article_url(Article.last), params: { article: { title: "updated" } }, xhr: true
 ```
 
-Now you should get a connection to remote browser.
+One more example: Calling the `:create` action to create a new article, passing
+in text for the `title` in `params`, as JSON request:
 
-```bash
-$ SELENIUM_REMOTE_URL=http://localhost:4444/wd/hub bin/rails test:system
+```ruby
+post articles_path, params: { article: { title: "Ahoy!" } }, as: :json
 ```
 
-If your application in test is running remote too, e.g. Docker container,
-Capybara needs more input about how to
-[call remote servers](https://github.com/teamcapybara/capybara#calling-remote-servers).
+NOTE: If you try running the `test_should_create_article` test from
+`articles_controller_test.rb` it will fail on account of the newly added model
+level validation and rightly so.
```

Comment:
A little simpler? Just a suggestion, feel free to ignore.
```suggestion
NOTE: If you try running the `test_should_create_article` test from
`articles_controller_test.rb` it will (correctly) fail due to the newly added model
level validation.
```